src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -1380,35 +1386,61 @@ SDCPUThread::sendToMemory(shared_ptr<InflightInst> inst_ptr,
     inst_ptr->notifyMemorying();
 
     weak_ptr<InflightInst> weak_inst(inst_ptr);
-    auto callback =
-        [this, weak_inst] (Fault fault) {
-            onExecutionCompleted(weak_inst, fault);
-        };
 
-    if (sreq) {
-        assert(sreq->high && sreq->low);
-        if (write) {
+    if (write) {
+        PacketPtr resp = Packet::createWrite(sreq ? sreq->main : req);
+        if (resp->hasData()) {
+            resp->dataStatic(data.get());
+        }
+        resp->makeResponse();
+
+        Fault f = inst_ptr->staticInst()->completeAcc(resp, inst_ptr.get(),
+                                                      inst_ptr->traceData());
+        // NOTE: SDCPU may do other things for special instruction types, which
+        //       may no longer work correctly with this change.
+
+        delete resp;
+
+        if (f != NoFault) {
```

Comment:
Should we have a panic here?

Reply:
looks like O3 ignores the return value of completeAcc() altogether :T

Reply:
My thought was that I've been assuming this never faults. I like to have an assert everywhere I make an assumption. But you're right, I *think* we'd handle a fault correctly in this case.